packages/functional/src/pipe.ts

```diff
@@ -20,26 +18,87 @@
  *      (y) => y + 1,
  * );
  * ```
- * @param init  The initial value
- * @param fns   Any number of functions to pipe the value through
- * @returns     The final value with all functions applied
+ *
+ * @see https://github.com/ramda/ramda/blob/master/source/pipe.js
+ * @see https://github.com/darky/rocket-pipes/blob/master/index.ts
+ *
+ * @example Basic
+ * ```ts
+ * const add = (a, b) => a + b;
+ * const add10 = x => add(x, 10);
+ * const add100 = x => add(x, 100);
+ * const sum = pipe(1, add10, add100);
+ * sum === 111; // true
+ * ```
+ *
+ * @example Building a Solana transaction message
+ * ```ts
+ * const transferTransactionMessage = pipe(
+ *     // The result of the first expression...
+ *     createTransactionMessage({ version: 0 }),
+ *     // ...gets passed as the sole argument to the next function in the pipeline.
+ *     tx => setTransactionMessageFeePayer(myAddress, tx),
+ *     // The return value of that function gets passed to the next...
+ *     tx => setTransactionMessageLifetimeUsingBlockhash(latestBlockhash, tx),
+ *     // ...and so on.
+ *     tx => appendTransactionMessageInstruction(createTransferInstruction(myAddress, toAddress, amountInLamports), tx),
+ * );
+ * ```
+ *
+ * @param init The initial value
+ * @returns The initial value
  */
 export function pipe<TInitial>(init: TInitial): TInitial;
+/**
+ * @param init The initial value
+ * @param init_r1 The function with which to transform the initial value
+ * @returns The return value of the final transform function
+ */
```

Comment:
I can not believe:

1. You have to document function arguments as `@param` in the docblock rather than inline if you want Intellisense to show them
2. You have to document every single overload; it won't just inherit from the top level

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/lE1zD7Jpg6mWNv3djNKZ/d640443a-1c24-4642-8348-0497f3a3f5e6.png)